Fb aux tools

aliabdolali opened the pull request:
# Pull Request Summary
## Description
Cloning WW3 to Windows machines is not possible as `aux` is a protected name.
This folder contains preprocessing tools (nml conversion/switch tools), so it is renamed to `tools`
What bug does it fix, or what feature does it add?
It allows Windows users to clone the code directly from github.
Is a change of answers expected from this PR?
No
### Issue(s) addressed
fixes #393 
### Check list  
* Is your feature branch up to date with the authoritative repository (NOAA/develop)?
Yes
* Reviewers: @JessicaMeixner-NOAA @thesser1 


### Testing
* How were these changes tested?
Regtest
* Are the changes covered by regression tests? (If not, why? Do new tests need to be added?)
NA
* If a new feature was added, was a new regression test added?
No
* Have regression tests been run?
Yes
* Which compiler / HPC you used to run the regression tests in the PR? 
```
intel/18.0.5.274
impi/2018.0.4
netcdf/4.7.4
```
* Please provide the summary output of matrix.comp (_matrix.Diff.txt_, _matrixCompFull.txt_ and _matrixCompSummary.txt_):  
[matrixDiff.txt](https://github.com/NOAA-EMC/WW3/files/6655793/matrixDiff.txt)
[matrixCompSummary.txt](https://github.com/NOAA-EMC/WW3/files/6655794/matrixCompSummary.txt)
[matrixCompFull.txt](https://github.com/NOAA-EMC/WW3/files/6655795/matrixCompFull.txt)
  
results were identical except the known list of non-identical tests
```
mww3_test_03
mww3_test_07
ww3_tp2.10
ww3_tp2.15
ww3_tp2.18
```

UKMO-lsampson committed:
Added boundary checks to the SMC grid input files for ww3_grid,
to ensure they comply with the limits of the nameslist.

ukmo-juan-castillo committed:
Fb 360 calendar (#8)

Changes to add support to 360-day and 365-day (no leap year) calendar - see ticket #209
  * Additional CALTYPE namelist parameter in MISC section
  * New ww3_tc1 regtest.

ukmo-ccbunney committed:
RTD support for ww3_boun[dc] (#10)

* Updated ww3_bound and ww3_bounc to handle model grids formulated on a rotated pole.
* Manual and nml/inp files to updated clarify that ww3_bound/ww3_bounc only accept input spectra formulated on a standard pole grid.

ukmo-juan-castillo committed:
Fb coupling time (#9)

Updates to allow a coupling time step that is different from the model time step. 
* Includes new regtest (in ww3_tp2.14) for non-default oasis time step.
* ww3_tp2.14 regtest added to matrix.base.

Commit:
Merge remote-tracking branch 'upstream/staging' into develop

Commit:
bug fix for ukmet development

Commit:
Merge remote-tracking branch 'upstream/develop' into develop

Commit:
Merge remote-tracking branch 'upstream/develop' into develop

Commit:
Merge remote-tracking branch 'upstream/develop' into develop

Commit:
Merge remote-tracking branch 'upstream/develop' into develop

Commit:
Merge remote-tracking branch 'upstream/develop' into develop

Commit:
Merge remote-tracking branch 'upstream/develop' into develop

Commit:
Merge remote-tracking branch 'upstream/develop' into develop

Commit:
Merge remote-tracking branch 'upstream/develop' into develop

Commit:
Merge remote-tracking branch 'upstream/develop' into develop

Commit:
aaRevert "Make ww3_grid into a callable subroutine (#362)"

This reverts commit 6efe9bb9c08207e71061d64082c8025cd97b9c61.

Commit:
Merge remote-tracking branch 'upstream/develop' into develop

Commit:
solve conflict

Commit:
rename aux to tools and change the name all over the code

JessicaMeixner-NOAA commented:
Is it just the folder aux that's the issue?  For example are the manual/*/aux.tex files okay? 

Doing a quick grep, it looks like things are okay.  There are a lot of mentions of "auxiliary" programs in the manual but, I think those are all fine to leave as is.

JessicaMeixner-NOAA commented:
@mickaelaccensi in looking at this PR,  I saw that there is a file: regtests/ww3_tp2.14/input/oasis3-mct/lib/mct/doc/.cvsignore should that be moved to .gitignore ?

mickaelaccensi commented:
> @mickaelaccensi in looking at this PR, I saw that there is a file: regtests/ww3_tp2.14/input/oasis3-mct/lib/mct/doc/.cvsignore should that be moved to .gitignore ?

yes it can be added to .gitignore

Commit:
remove aux.tex from manual

aliabdolali commented:
> Is it just the folder aux that's the issue? For example are the manual/*/aux.tex files okay?
> 
> Doing a quick grep, it looks like things are okay. There are a lot of mentions of "auxiliary" programs in the manual but, I think those are all fine to leave as is.

Good catch @JessicaMeixner-NOAA 
I renamed it to sup.tex and tried to clone WW3 to a Windows machine I have access to.
Now, another protected name pops up. (model/esmf/con.F90). Can we go ahead and rename this and modify the code wherever it is mentioned? do you have any preferences?

Commit:
add .cvsignore for regtests/ww3_tp2.14 in .gitignore

aliabdolali commented:
> > @mickaelaccensi in looking at this PR, I saw that there is a file: regtests/ww3_tp2.14/input/oasis3-mct/lib/mct/doc/.cvsignore should that be moved to .gitignore ?
> 
> yes it can be added to .gitignore

Just added, thanks

JessicaMeixner-NOAA commented:
@aliabdolali I think it's find to rename model/esmf/con.F90 to a different name that will allow WW3 to be cloned on a windows machine.  I don't have a preference.  You could email Tim if you want his input since he's the original developer.  Based off https://github.com/NOAA-EMC/WW3/blob/develop/model/esmf/con.F90#L3 I'd say maybe "concomp"  if you are looking for a suggestion.

Commit:
Merge remote-tracking branch 'upstream/develop' into fb_aux_tools

Commit:
rename con.F90 to concomp.F90 and releted names for esmf

aliabdolali commented:
> @aliabdolali I think it's find to rename model/esmf/con.F90 to a different name that will allow WW3 to be cloned on a windows machine. I don't have a preference. You could email Tim if you want his input since he's the original developer. Based off https://github.com/NOAA-EMC/WW3/blob/develop/model/esmf/con.F90#L3 I'd say maybe "concomp" if you are looking for a suggestion.

@JessicaMeixner-NOAA I checked with Tim and after his confirmation, renamed them and modified makefile and tested the compilation of ww3_multi_esmf successfully. 
Now, I am running the full regrest
I also cloned WW3 on a windows machine with no problem.

Commit:
Merge remote-tracking branch 'upstream/develop' into fb_aux_tools

aliabdolali commented:
The tests are done successfully, with know non b4b tests.
```
mww3_test_03
mww3_test_07
ww3_tp2.10
ww3_tp2.15
```
[matrixDiff.txt](https://github.com/NOAA-EMC/WW3/files/6683788/matrixDiff.txt)
[matrixCompSummary.txt](https://github.com/NOAA-EMC/WW3/files/6683789/matrixCompSummary.txt)
[matrixCompFull.txt](https://github.com/NOAA-EMC/WW3/files/6683790/matrixCompFull.txt)

Please provide your final review and it is ready to be integrated.

JessicaMeixner-NOAA commented on .gitignore:
```diff
@@ -63,6 +63,7 @@ regtests/ww3_tp2.17/input/inlet.msh
 regtests/ww3_tp2.21/input/mesh.msh
 regtests/ww3_tp2.21/input/obstructions_local.glo_unst.in
 regtests/ww3_tp2.21/input/obstructions_shadow.glo_unst.in
+regtests/ww3_tp2.14/input/oasis3-mct/lib/mct/*/.cvsignore
```
I don't see where this file was renamed from .cvsignore to .gitignore or deleted (https://github.com/NOAA-EMC/WW3/blob/develop/regtests/ww3_tp2.14/input/oasis3-mct/lib/mct/.cvsignore) so I don't know how adding this to the gitignore is helpful?

JessicaMeixner-NOAA replied:
There are also a couple of other .cvsignore files in the regtests/ww3_tp2.14/input/oasis3-mct directories, maybe we just leave this out of this PR since it's not actually related and let @mickaelaccensi  clean this up in the future?

JessicaMeixner-NOAA reviewed (approved):
I'm approving although, I do think that .gitignore updates should probably be reverted and just take care of that separately.

Commit:
revert ww3_tp2.14/input/oasis3-mct/lib/mct/*/.cvsignore from gitignore
